lti_consumer/lti_xblock.py

```diff
@@ -264,6 +266,15 @@ class LtiConsumerXBlock(StudioEditableXBlockMixin, XBlock):
             "LTI 1.3 and LTI Advantage features."
         ),
     )
+
+    external_config = String(
+        display_name=_("External configuration"),
+        scope=Scope.settings,
+        values=get_external_config_values,
+        help=_("Select the External configuration for the LTI Tool")
+    )
```

Comment:
@giovannicimolin While working on removing direct dependencies between the XBlock and the Filters, I got held up here. Here `get_external_config_values` allows the XBlock to fetch a list of available LTI configurations from plugins to display it to the user in the "Edit" page.

Considering your other comment about passing `course_id` to the filters, we can't use this function reference to `values` anymore. So I am thing of adding a AJAX XBlock handler, that will allow the front-end to fetch the list of available LTI Tools. This will allow the XBlock to pass the `course_id` to the `get_external_config_values` function.

But, this still means, there will be a link between XBlock and the Filters. Do you have any idea about how we can get this list without XBlock depending on Filters?

Reply:
Ah, I think I understand the problem now. 

What I had in mind for this field was to have it as a simple `string` field, and [modify the context](https://github.com/openedx/xblock-lti-consumer/blob/master/lti_consumer/lti_xblock.py#L942) or the [method that prepares the fields](https://github.com/openedx/xblock-utils/blob/master/xblockutils/studio_editable.py#L96) to fool the template into rendering a select instead of text input. But I think this is a bit too much for the first iteration.

Let's avoid the complexity of dealing with the XBlock studio view rendering for now: make this a simple `String` field.
Make sure this is gated behind a WaffleFlag, so the option and associated field don't appear if it's globally disabled (use something like this: https://github.com/openedx/xblock-lti-consumer/commit/92cf504eba66577103e3bd6650bdc58a9fa9ffa9).

Reply:
@giovannicimolin Oh! I too tried fooling the system and failed :) I populated the `values` for the field in the `studio_view` before creating the fragment, but somehow the change never persisted. I didn't use any of the XBlock's internal functions though, simply updated the `values` list. Let's keep it simply as a string field as you say. We will get to this UX part later :)